src/net.rs

```diff
@@ -172,10 +173,23 @@ impl Builder {
         self
     }
 
+    /// Set the ALPN this gossip instance uses.
+    ///
+    /// It has to be the same for all peers in the network. If you set a custom ALPN,
+    /// you have to use the same ALPN when registering the [`Gossip`] in on a iroh
+    /// router with [`RouterBuilder::accept`].
+    ///
+    /// [`RouterBuilder::accept`]: iroh::protocol::RouterBuilder::accept
+    pub fn alpn(mut self, alpn: impl AsRef<[u8]>) -> Self {
```

Comment:
We could use impl Into<Bytes> here, that also covers a lot and would allow passing in a Bytes.

<img width="198" height="242" alt="image" src="https://github.com/user-attachments/assets/1e96e91c-43bc-4f11-b32e-310f5eab1a08" />

Reply:
Left it at `impl AsRef<[u8]>` for now (same as `RouterBuilder::accept`) becaues otherwise it won't take `b"my-alpn"` because that is `&'static [u8; N]` and not `&'static [u8]` unfortunately.